lib/tasks/evm_dba.rake

```diff
@@ -18,6 +18,55 @@ module EvmDba
     return false unless config['adapter'] == 'postgresql'
     return %w( 127.0.0.1 localhost ).include?(config['host']) || config['host'].blank?
   end
+
+  def self.with_options(*option_types, &block)
+    require 'trollop'
+
+    Trollop.options(EvmRakeHelper.extract_command_options) do
+      option_types.each do |type|
+        case type
+        when :db_credentials
+          opt :username,         "Username",                   :type => :string
+          opt :password,         "Password",                   :type => :string
+          opt :hostname,         "Hostname",                   :type => :string
+          opt :dbname,           "Database name",              :type => :string
+        when :local_file
+          opt :local_file,       "Destination file",           :type => :string, :required => true
+        when :remote_file
+          opt :remote_file_name, "Destination depot filename", :type => :string
+        when :splitable
+          opt :byte_count,        "Size to split files by",     :type => :string
+        when :remote_uri
+          opt :uri,              "Destination depot URI",      :type => :string, :required => true
+          opt :uri_username,     "Destination depot username", :type => :string
+          opt :uri_password,     "Destination depot password", :type => :string
+        when :exclude_table_data
+          opt :"exclude-table-data", "Tables to exclude data", :type => :strings
```

Comment:
I don't understand this hyphenization.  Trollop already auto-hypenates things.

Reply:
> Trollop already auto-hypenates things.

Oh, I guess I didn't know that.  That said, I might have done this on purpose because `PostgresAdmin` passes this option down to `pg_dump` via `AwesomeSpawn`, and that does not `auto-hypenate`, if I am not mistaken...

... _Nick goes to check_...

Reply:
A few things of note about this (since it is a good spot where I can leave this comment, reference it, and no one will get on my case about it being a :book:...)

- So `Trollop` does auto-hyphenate as mentioned, but it does not do that when you pass the `flag` option directly.
- The above is an important distinction because we are auto generating our flags, even our short ones.  Those `short_opts`, because of the `EvmDba.with_options` changes might change as we add/remove new opts over time.
- Because of the short opt generation, we should probably consider one of following:
  - In internal tools, never use short opts when calling out to these tasks from elsewhere (i.e. `appliance_console`)
  - Hard coding our short opts
  - Disabling our short opts

Also, because I am a "troll", this is why I just use `optparse`.  :trollface: